File: src/Shell/Module1.vb
Imports System.IO
Imports System.Net
Imports System.Net.Sockets

Module Module1

    Dim _client As TcpClient

    Sub Main(ByVal args() As String)
        _client = New TcpClient(args(0), 5555)

        Dim d As New Action(AddressOf ReadCon)
        d.BeginInvoke(Nothing, Nothing)

        Dim input As String
        Dim sw As New StreamWriter(_client.GetStream)

        Do
            input = Console.ReadLine
            If input = "exit" Then Exit Sub

            sw.Write(input & vbCrLf)
            sw.Flush()
            If Not _client.Connected Then Exit Sub
        Loop
    End Sub

    Sub ReadCon()
        Dim sr As New StreamReader(_client.GetStream)

        While _client.Connected
            Console.WriteLine(sr.ReadLine)
        End While
    End Sub
End Module

File: src/sbsh/Module1.cs
using System;
using System.IO;
using System.Net.Sockets;

namespace Pixa.Soundbridge.Shell
{
    static class Module1
    {
        private static TcpClient _client;

        public static void Main(string[] args)
        {
            _client = new TcpClient(args[0], 5555);
            var d = new Action(ReadCon);
            d.BeginInvoke(null, null);
            string input;
            var sw = new StreamWriter(_client.GetStream());
            do
            {
                input = Console.ReadLine();
                if (input == "exit")
                    return;
                sw.Write(input + "\r\n");
                sw.Flush();
                if (!_client.Connected)
                    return;
            }
            while (true);
        }

        public static void ReadCon()
        {
            var sr = new StreamReader(_client.GetStream());
            while (_client.Connected)
                Console.WriteLine(sr.ReadLine());
        }
    }
}